src/iftraf.c

```diff
@@ -0,0 +1,1051 @@
+#include <stdio.h>
+#include <string.h>
+#include "inet.h"
+#include "ipv4.h"
+#include "ipv6.h"
+#include "iftraf.h"
+#include "conf/iftraf.h"
+
+#ifndef IFTRAF
+#define IFTRAF
+#define RTE_LOGTYPE_IFTRAF    RTE_LOGTYPE_USER1
+#endif
+
+#define IFTRAF_TOPN 20
+
+#define IFTRAF_RING_SIZE  1024
+#define IFTRAF_INTERVAL   1024 
+
+#define IFTRAF_PKT_DIR_IN 0
+#define IFTRAF_PKT_DIR_OUT 1
+
+#define IFTRAF_TBL_BITS          12
+#define IFTRAF_TBL_SIZE          (1 << IFTRAF_TBL_BITS)
+#define IFTRAF_TBL_MASK          (IFTRAF_TBL_SIZE - 1)
+
+#define IFTRAF_IFTBL_BITS          10
+#define IFTRAF_IFTBL_SIZE          (1 << IFTRAF_IFTBL_BITS)
+#define IFTRAF_IFTBL_MASK          (IFTRAF_IFTBL_SIZE - 1)
+
+#define HISTORY_LENGTH  30
+int history_pos = 0;
+
+int ticket = 0;
+#define TIME_INTERVAL 0x100000
+
+bool iftraf_disable = true;
+
+typedef struct sorted_list_node_tag {
+    struct sorted_list_node_tag* next;
+    void* data;
+} sorted_list_node;
+
+typedef struct {
+    sorted_list_node root;
+    int (*compare)(void*, void*);
+} sorted_list_type;
+
+int sorted_list_num = 0;
+sorted_list_type iftraf_sorted_list;

```

Comment:
1. Place Macro "TIME_INTERVAL" before variable definitions.
2. Use "static" global variables if they are used in this file only.
3. Use module name prefix for global naming to reduce conflict chances, such as: "ticket"->"iftraf_ticket", "TIME_INTERVAL"->"IFTRAF_TIME_INTERVAL", ...